### ISSUE TITLE
 Greasemonkey 3.0: GM_ functions don't work when called with `apply` 

### ISSUE THREAD

**Issue by arantius** [greasemonkey/greasemonkey#2129]
Source: https://groups.google.com/d/topic/greasemonkey-dev/JpF9COGzjnQ

> In the past, I've been calling the Greasemonkey functions inside a wrapper like this:
>
>      GM_registerMenuCommand.apply(rootScope, arguments);
>
> Upgrading to 3.0 has broken this. Now those calls either never finish or silently error (hard to tell which).
>
> Is this intended behavior? If so, is there another way to apply a variable number of arguments to the Greasemonkey functions?

**Comment by arantius** [comment]
Script:

```javascript
// ==UserScript==
// @name        API apply() test
// @namespace   x
// @description x
// @include     http*
// @version     1
// @grant       GM_log
// ==/UserScript==

GM_log('bare GM_log');
GM_log.apply(this, ['bare GM_log']);
```

Firefox 37.0.2.  GM 3.1, confirmed: `Permission denied to access property "apply"`
Firefox 37.0.2, GM 2.3, confirmed: `Permission denied to access property "apply"`
Firefox 30.0, GM 2.3, works.

This is not new in 3.x.  This is a change in Firefox.

**Comment by arantius** [comment]
Firefox 34.0.5, GM 2.3, works.
 Firefox 36.0.4, GM 2.3, fails.
 Firefox 35.0.1, GM 2.3, works.
 Firefox 35.0.1, GM 3.1, works.
 
 Change between Firefox 35 and 36.  Unrelated to GM 3.x.

**Comment by arantius** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1163019

**Comment by janekptacijarabaci** [comment]
Firefox 35.0.1, GM 3.x - works. However, there is help:

![1](https://cloud.githubusercontent.com/assets/2373486/7542974/4ef5105e-f5c1-11e4-9e41-9ff4e84b4650.png)

Firefox 36.0beta1, GM 3.x, fails

**Comment by janekptacijarabaci** [comment]
__Ad Firefox 35.0__

**_WORKS_**

vs.

**_"WORKS"_**
GM_log.apply() - Throws an warning in the Browser Console:
*"Security wrapper denied access to property apply on privileged Javascript object. Support for exposing privileged objects to untrusted content via \__exposedProps__ is being gradually removed - use WebIDL bindings or Components.utils.cloneInto instead. Note that only the first denied property access from a given global object will be reported."*

__Last good: 35.0a1 (2014-10-03)__
Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0
Built from https://hg.mozilla.org/mozilla-central/rev/b85c260821ab

__First bad: 35.0a1 (2014-10-04)__
Mozilla/5.0 (Windows NT 6.1; WOW64; rv:35.0) Gecko/20100101 Firefox/35.0
Built from https://hg.mozilla.org/mozilla-central/rev/229801d17f7e

Pushlog
http://hg.mozilla.org/mozilla-central/pushloghtml?fromchange=b85c260821ab&tochange=229801d17f7e

[Bug 1065185](https://bugzilla.mozilla.org/show_bug.cgi?id=1065185) is the suspect.
__________

__Ad Firefox 36.0__

**_WORKS_**

GM_log.apply() - Throws an warning in the Browser Console:
*"Security wrapper denied access to property apply on privileged Javascript object. Support for exposing privileged objects to untrusted content via \__exposedProps__ is being gradually removed - use WebIDL bindings or Components.utils.cloneInto instead. Note that only the first denied property access from a given global object will be reported."*

vs.

**_FAILS_**

Throws an error in the Browser Console:
Permission denied to access property "apply"

__Last good: 36.0a1 (2014-10-20)__
Mozilla/5.0 (Windows NT 6.1; WOW64; rv:36.0) Gecko/20100101 Firefox/36.0
https://hg.mozilla.org/mozilla-central/rev/33c0181c4a25

__First bad: 36.0a1 (2014-10-21)__
Mozilla/5.0 (Windows NT 6.1; WOW64; rv:36.0) Gecko/20100101 Firefox/36.0
https://hg.mozilla.org/mozilla-central/rev/29fbfc1b31aa

Pushlog
http://hg.mozilla.org/mozilla-central/pushloghtml?fromchange=33c0181c4a25&tochange=29fbfc1b31aa

[Bug 1081990](https://bugzilla.mozilla.org/show_bug.cgi?id=1081990) is the suspect.
[Bug 1081985](https://bugzilla.mozilla.org/show_bug.cgi?id=1081985) is the suspect.

**Comment by janekptacijarabaci** [comment]
See also https://github.com/greasemonkey/greasemonkey/issues/2097

**Comment by janekptacijarabaci** [comment]
[Bug 1163019](https://bugzilla.mozilla.org/show_bug.cgi?id=1163019) RESOLVED WONTFIX (see also #2356)

**Comment by janekptacijarabaci** [comment]
See also https://github.com/greasemonkey/greasemonkey/issues/2097
